### ISSUE TITLE
Support for "compressed" .framework references

### ISSUE THREAD

**Issue by mattleibow** [xamarin/xamarin-macios#11671]
### Steps to Reproduce

1. Create a Mac Catalyst .framework
2. Add it to a NuGet package

### Expected Behavior

The framework does not get corrupted.

### Actual Behavior

The compression of a NuGet uses a compression system that does not preserve symbolic links, causing compiler failures.

### Environment

.NET 6

**Comment by filipnavara** [comment]
This is actually NuGet issue so maybe https://github.com/NuGet/Home would be better fit?

**Comment by spouliot** [comment]
> This is actually NuGet issue so maybe https://github.com/NuGet/Home would be better fit?

Agreed. Any workaround will have non negligible impact on build performance.

**Comment by mattleibow** [comment]
Adding a link to that: https://github.com/NuGet/Home/issues/10893

But I don't think this is a nuget issue. Since this is really an issue with the Windows OS not supporting symbolic links properly, the SDK will have to handle this.

**Comment by mattleibow** [comment]
Adding a link to another, more generic issue: https://github.com/NuGet/Home/issues/10734

**Comment by spouliot** [comment]
> Since this is really an issue with the Windows OS not supporting symbolic links properly, the SDK will have to handle this.

Recently there was an XML file added to UNIX-like permission could be kept properly.

**Comment by spouliot** [comment]
We need to verify if macOS (and Catalyst) support [0] the newer "user framework" format that is used by iOS, tvOS and watchOS. Those frameworks are not using symlinks (since they do not have to care about versioning).

If this is supported then the frameworks could be converted to this format - and that would solve the issue.

[0] build time, run time and submissions to App Store

**Comment by rolfbjarne** [comment]
This has been implemented, we'll automatically zip up any (xc)frameworks with symlinks now.
